src/index.ts

```diff
@@ -80,13 +80,22 @@ export const traceProvider: TraceProvider = {
   },
 
   /**
-   * @description Used to propagate traces. Accepts a serialised trace
-   * metadata string and returns a traceContext instance that can be passed
-   * to the wrapHandler.
+   * @description Used to propagate traces. Accepts serialised trace
+   * metadata and returns a traceContext instance that can be passed
+   * to wrapHandler.
    */
   deserializeTraceMetadata: async (
-    traceMetadata: string
-  ): Promise<unknown> => ({
-    distributedTraceHeaders: JSON.parse(traceMetadata),
-  }),
+    traceMetadata: unknown
+  ): Promise<unknown> => {
+    try {
+      if (typeof traceMetadata !== "string") {
+        return {};
+      }
+      return {
+        distributedTraceHeaders: JSON.parse(traceMetadata),
```

Comment:
Do we need to do any validation on the object before returning it?

Reply:
The `traceMetadata` itself is provided by New Relic, so I'm treating it here as opaque data. That being said, we do know the expected keys (traceparent, tracestate, etc.), so we _could_ do a deeper inspection (will add a TODO for this). If we pass an invalid value to New Relic it does gets silently ignored, so it's low-risk.